packages/astro/src/runtime/server/render/slot.ts

```diff
@@ -35,7 +35,7 @@ export async function* renderSlot(
 		yield* iterator;
 	}
 
-	if (fallback) {
+	if (fallback && !slotted) {
```

Comment:
The `yield` keyword doesn't return so we also have to check that there's no slotted content